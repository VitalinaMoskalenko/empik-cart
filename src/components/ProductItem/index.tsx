import React from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";
import { H3 } from "../Headings";
import { Body } from "../Paragraphs";

type PropsType = {
  title: string;
  price: string;
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const Price = styled(Body)`
  font-weight: bold;
`;

const CurrencyPrice = styled.span`
  color: ${({ theme }) => theme.colors.orange};
`;

const baseTranslationPath = "Components.ProductItem.";

const ProductItem = ({ title, price }: PropsType) => {
  const { t } = useTranslation();

  return (
    <Container>
      <H3>{title}</H3>
      <Price>
        {t(`${baseTranslationPath}price`)}
        <CurrencyPrice>
          {t(`${baseTranslationPath}currency`, { price })}
        </CurrencyPrice>
      </Price>
    </Container>
  );
};

export default ProductItem;
