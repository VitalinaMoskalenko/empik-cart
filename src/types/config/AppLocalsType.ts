export enum AppLocalsType {
  pl = "pl",
}
