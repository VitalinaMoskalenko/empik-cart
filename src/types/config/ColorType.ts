export enum ColorType {
  white = "white",
  black = "black",
  grey = "grey",
  orange = "#ff6903",
}
