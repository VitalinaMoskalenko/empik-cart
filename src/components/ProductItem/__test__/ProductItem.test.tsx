import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import ProductItem from "../";
import { theme } from "../../../config";

it("renders ProductItem correctly", () => {
  const productItemComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ProductItem title="Title" price="10" />
      </ThemeProvider>
    )
    .toJSON();

  expect(productItemComponent).toMatchSnapshot();
});
