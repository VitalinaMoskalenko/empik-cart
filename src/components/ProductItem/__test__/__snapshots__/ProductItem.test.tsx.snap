// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductItem correctly 1`] = `
<div
  className="sc-iCoGMd bDrLOR"
>
  <h1
    className="sc-dlnjwi czuNcg"
  >
    Title
  </h1>
  <p
    className="sc-jSFjdj sc-fujyAs bQuzKy lpvfqg"
  >
    Components.ProductItem.price
    <span
      className="sc-pNWdM fxxTmU"
    >
      Components.ProductItem.currency
    </span>
  </p>
</div>
`;
